### ISSUE TITLE
Add support for global excluding of transitive dependencies

### ISSUE THREAD

**Issue by Gedochao** [VirtusLab/scala-cli#3354]
**Is your feature request related to a problem? Please describe.**
We'd like a way to globally exclude transitive dependencies, rather than doing it via individual dep param (`exclude=*`, as described in https://scala-cli.virtuslab.org/docs/guides/introduction/dependencies/#excluding-transitive-dependencies).

**Describe the solution you'd like**
- [ ] `using` directive: `//> using dep.exclude org.scala-lang.modules::scala-collection-compat:2.12.0`  
- [ ] command line option: `--dependency-exclude org.scala-lang.modules::scala-collection-compat:2.12.0`
- [ ] passing the dependency without an explicit version should exclude it independent of the transitive version, i.e.: `//> using dep.exclude org.scala-lang.modules::scala-collection-compat` would get rid of all transitive `scala-collection-compat`

**Describe alternatives you've considered**
Keep excluding per dependency, when necessary.

**Additional context**
Initially suggested by @prolativ.

**Comment by tgodzik** [comment]
Not sure if we have actual way to exclude a specific version, does coursier have an API for that?

**Comment by prolativ** [comment]
This should also work with other variants of `dep`, e.g. `compileOnly.dep.exclude` or `test.dep.exclude`

**Comment by prolativ** [comment]
Shouldn't the command line option be named `--dep-exclude`? Or maybe we should at least have such an alias?
In order to keep symmetry with `--compile-dep`, while `--compile-dependency` is missing

**Comment by Gedochao** [comment]
> Shouldn't the command line option be named `--dep-exclude`? Or maybe we should at least have such an alias? In order to keep symmetry with `--compile-dep`, while `--compile-dependency` is missing

yeah, we should have an alias in compliance with `--dep`/`--dependency`.
I suppose `--compile-dependency` is a missing alias, here.
